diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -138,12 +143,31 @@ fun ASTNode.isFollowedByNewline() =
             it.treeNext.elementType == WHITE_SPACE && it.treeNext.text.contains("\n")
         } ?: false
 
+/**
+ * This function is similar to isFollowedByNewline(), but there may be a comment after the node
+ */
+fun ASTNode.isFollowedByNewlineWithComment() =
+        parent({ it.treeNext != null }, strict = false)?.let { astNode ->
+            astNode.treeNext.run {
+                when (elementType) {
+                    WHITE_SPACE -> text.contains("\n")
+                    EOL_COMMENT, BLOCK_COMMENT, KDOC -> isFollowedByNewline()
+                    else -> false
+                } ||
+                        parent({ it.treeNext != null }, strict = false)?.let {
+                            it.treeNext.elementType == EOL_COMMENT && it.treeNext.isFollowedByNewline()
+                        } ?: false
+            }
+        } ?: false
+
 /**
  * Checks if there is a newline before this element. See [isFollowedByNewline] for motivation on parents check.
+ * Or if there is nothing before, it cheks, that there are empty imports and package before (Every FILE node has children of type IMPORT_LIST and PACKAGE)
  */
 fun ASTNode.isBeginByNewline() =
         parent({ it.treePrev != null }, strict = false)?.let {
-            it.treePrev.elementType == WHITE_SPACE && it.treePrev.text.contains("\n")
+            it.treePrev.elementType == WHITE_SPACE && it.treePrev.text.contains("\n") ||
+                    (it.treePrev.elementType == IMPORT_LIST && it.treePrev.isLeaf() && it.treePrev.treePrev.isLeaf())
```

Comment:
Is `treePrev.treePrev` always not null here? Can it be guaranteed or should be checked?

Reply:
Yes, because each `FILE` node contains `PACKAGE_DIRECTIVE` before `IMPORT_LIST`